sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -480,11 +480,21 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
     try {
       Option(hive.getFunction(db, name)).map(fromHiveFunction)
     } catch {
-      case CausedBy(ex: NoSuchObjectException) if ex.getMessage.contains(name) =>
+      case e: Throwable if isCausedBy(e, s"$name does not exist") =>
```

Comment:
yes, but would you mind doing `case NonFatal(e) =>` here? It's generally a bad practice to catch throwables